mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -2296,6 +2297,184 @@ impl TendermintCoin {
             memo: Some(req.memo),
         })
     }
+
+    pub(crate) async fn undelegate(&self, req: DelegationPayload) -> MmResult<TransactionDetails, DelegationError> {
+        fn generate_message(
+            delegator_address: AccountId,
+            validator_address: AccountId,
+            denom: Denom,
+            amount: u128,
+        ) -> Result<Any, String> {
```

Comment:
Could we return the original ErrorReport instead of String here and implement `impl From<ErrorReport> for DelegationError`?
(sorry it's my dislike of 'String' as errors)